addons/delivery/models/res_partner.py

```diff
@@ -1,14 +1,65 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import fields, models
+from odoo import api, fields, models
 from odoo.fields import Domain
 
 
 class ResPartner(models.Model):
     _inherit = 'res.partner'
 
     property_delivery_carrier_id = fields.Many2one('delivery.carrier', company_dependent=True, string="Delivery Method", help="Used in sales orders.")
-    is_pickup_location = fields.Boolean()  # Whether it is a pickup point address.
+    pickup_carrier_id = fields.Many2one('delivery.carrier')  # The shipping connector that generated this pickup location
```

Comment:
why do we need to save the whole delivery method while we can save only the code which is lighter?
```suggestion
    pickup_delivery_method_code = fields.Many2one('delivery.carrier')  # The delivery method code that generated this pickup location.
```